app/services/exchanger_service.rb

```diff
@@ -0,0 +1,29 @@
+class ExchangerService
+  def initialize(params)
+    @params = params
+  end
+
+  def call(currency = CurrencyService.new(params[:date]))
```

Comment:
So this is a bit confusing. So you're making currency optional, but you use CurrencyService methods directly below, which means it is actually not a option to input anything else, if not it'll cause an error. 

Also you've a model called Currency, so if you want to use the variable name it should be an instance of currency or it'll get confusing fast.